### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by AleksandarFilipov**
Major code quality improvment

* Share codepath for signing requests
* Move testcode to main test file
* State replicated in holder is not initiated by holder for odd cases. Such as intentional pid kill
* Significant renaming for the better
* Round initiator is more consistent

### CODE REVIEW

File: apps/ae_socket_connector/test/aesocketconnector_test.exs [hunk 1]
```diff
@@ -1,6 +1,7 @@
 defmodule SocketConnectorTest do
```
Comment (tolbrino) [inline apps/ae_socket_connector/test/aesocketconnector_test.exs]: This file should use the underscores in the name too.

File: README.md [hunk 2]
```diff
@@ -55,7 +55,7 @@ Start the samples at your prompt by doing
 mix test
 ```
 
-Scenarios executed can be found [here](apps/ae_socket_connector/test/scenarios_test.exs)
+Scenarios executed can be found [here](apps/ae_socket_connector/test/aesocketconnector_test.exs)
```
Comment (tolbrino) [inline README.md]: See comment below regarding filename.

**Review by tolbrino — CHANGES_REQUESTED** [review]

File: apps/ae_socket_connector/test/aesocketconnector_test.exs (hunk 1 above)
Comment (AleksandarFilipov) [reply apps/ae_socket_connector/test/aesocketconnector_test.exs]: fixed

File: README.md (hunk 2 above)
Comment (AleksandarFilipov) [reply README.md]: done

**Review by tolbrino — APPROVED** [review]

**Pull request merged**